src/main/java/htsjdk/samtools/util/CigarUtil.java

```diff
@@ -201,6 +212,13 @@ public static void clip3PrimeEndOfRead(SAMRecord rec, final int clipFrom, final
             }
         }
 
+        if (rec.getReadLength() != originalReadLength) {
+            //invalidate NM, UQ, MD tags if we have changed the length of the read.
+            rec.setAttribute(SAMTag.NM.name(), null);
```

Comment:
Add a test that shows these get invalidated.

Reply:
done